.github/workflows/build.yml

```diff
@@ -97,3 +101,56 @@ jobs:
           meson setup build
           meson compile -C build
           meson test -C build
+
+  msys-meson:
+    name: 'msys-meson ${{ matrix.sys.abi }} ${{ matrix.library }}'
+    runs-on: windows-latest
+    defaults:
+      run:
+        shell: msys2 {0}
+    strategy:
+      matrix:
+        sys:
+          - { abi: mingw32, env: i686,         compiler: gcc }
+          - { abi: mingw64, env: x86_64,       compiler: gcc }
+          - { abi: ucrt64,  env: ucrt-x86_64,  compiler: gcc }
+          - { abi: clang32, env: clang-i686,   compiler: clang }
+          - { abi: clang64, env: clang-x86_64, compiler: clang }
+        library: ['shared', 'static']
+      fail-fast: false
+    steps:
+      - name: Use MinGW from MSYS
+        uses: msys2/setup-msys2@v2
+        with:
+          msystem: ${{matrix.sys.abi}}
+          update: true
+          path-type: minimal
+          install: >-
+            mingw-w64-${{matrix.sys.env}}-toolchain
+            mingw-w64-${{matrix.sys.env}}-meson
+            mingw-w64-${{matrix.sys.env}}-ninja
```

Comment:
Just use the ["pacboy" `foo:p` way of installing packages for the matching environment,](https://github.com/msys2/setup-msys2#pacboy) there is no real need to carry around this `env` variable...

Also, the `toolchain` metapackage will pull in _a lot_ of unnecessary stuff and waste CI resources (both time and watts), please use the minimal set of packages needed - [for lcms2 this are just `cc` and `pkgconf` (w/ `meson` and `ninja` of course).](https://packages.msys2.org/package/mingw-w64-ucrt-x86_64-lcms2?repo=ucrt64)

Reply:
In this context, `pacboy` syntax will not work as the Action directly invokes pacman. `toolchain` here is used for reduced brittleness because you don't just need a compiler and pkg-config, you also need a pile of other dev tools for a correct environment such as various headers that are not installed by default. Which ones depends on the sub-environment, and `toolchain` guarantees you pull the correct things in.

Reply:
I don't think so, I have been using both pacboy and minimal packages needed for build in several CI actions for various projects already. Please try both of these out.